src/crypto/library/keccak/Keccak.cpp

```diff
@@ -0,0 +1,164 @@
+//
+// Created by PetterFi on 28.07.22.
+//
+// based on
+// https://github.com/nayuki/Bitcoin-Cryptography-Library
+
+#include "0_stdafx.hpp"                      // IWYU pragma: associated
+#include "1_Internal.hpp"                    // IWYU pragma: associated
+#include "crypto/library/keccak/Keccak.hpp"  // IWYU pragma: associated
+
+#include "internal/crypto/library/Factory.hpp"
+#include "internal/util/LogMacros.hpp"
+#include "opentxs/crypto/library/HashingProvider.hpp"
+
+#include <cassert>
+
+using std::size_t;
+using std::uint8_t;
+using std::uint64_t;
+
+namespace opentxs::factory
+{
+auto Keccak() noexcept -> std::unique_ptr<crypto::Keccak>
+{
+    using ReturnType = crypto::implementation::Keccak;
+
+    return std::make_unique<ReturnType>();
+}
+}  // namespace opentxs::factory
+
+namespace opentxs::crypto::implementation
+{
+Keccak::Keccak() noexcept {}
+
+auto Keccak::Digest(
+    const crypto::HashType hashType,
+    const ReadView data,
+    const AllocateOutput output) const noexcept -> bool
+{
+    try {
+        if (false == output.operator bool()) {
+            throw std::runtime_error{"invalid output"};
+        }
+
+        const auto size = HashSize(hashType);
+        auto buf = output(size);
+
+        if (false == buf.valid(size)) {
+            throw std::runtime_error{"failed to allocate space for output"};
+        }
+
+        auto* uint_data_array = new uint8_t[HASH_CHAR_NUMBER];
+        memcpy(
+            uint_data_array,
+            reinterpret_cast<const char*>(data.data()),
+            HASH_CHAR_NUMBER);
+
+        getHash(uint_data_array, data.size(), buf.as<uint8_t>());
+        return true;
+    } catch (const std::exception& e) {
+        LogError()(OT_PRETTY_CLASS())(e.what())("Cannot keccak hash").Flush();
+
+        return false;
+    }
+}
+
+auto Keccak::HMAC(
+    const crypto::HashType hashType,
+    const ReadView key,
+    const ReadView data,
+    const AllocateOutput output) const noexcept -> bool
+{
+    return false;
+}
+
+void Keccak::getHash(
```

Comment:
No I don't, I copied only a necessary code from a third party library: https://github.com/nayuki/Bitcoin-Cryptography-Library. When I was implementing it there was no cryptopp in the code. It was supposed to be a temporary solution because the final solution I wanted is to upgrade to OpnenSSL3 and use Keccak from this library. Probably I could check this cryptopp and replace the code.

Reply:
I have arranged with Jakub that for now, we will push this single header keccak256 implementation but later we will try to build CryptoPP from sources and put them in deps both for the libcsprrpc and for the opentxs.